File: BackEnd/app/src/main/java/app/app/entity/Carro.java
package app.app.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Carro {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private String nome;
    private int ano;

    //cria nao somente o carro, mas a marca tambem se ela nao existir
    @ManyToOne(cascade = CascadeType.ALL)
    private Marca marca;

    @ManyToMany(cascade = CascadeType.ALL)
    @JoinTable(name = "carro_proprietario")
    private List<Proprietario> proprietarios;
}

File: frontend/src/app/models/carro.ts
import { Marca } from './marca';
import { Proprietario } from './proprietario';

export class Carro {
  id!: number;
  nome!: string;
  ano!: number;
  marca!: Marca;
  proprietarios!: Proprietario[];
  valorFIPE!: number;
}
